PayPal決済APIサポートを追加

wamazing-sei-sato opened the pull request:
## 概要
GMO-PGのPayPal決済APIに対応するため、必要なメソッドを追加しました。

## 関連Issue
- https://github.com/wamazing/wamazing/issues/3486

## 変更内容

### 新規追加メソッド
- `entry_tran_paypal`: PayPal決済取引登録
- `exec_tran_paypal`: PayPal決済実行  
- `cancel_tran_paypal`: PayPal決済キャンセル
- `cancel_auth_paypal`: PayPal仮売上取消
- `paypal_sales`: PayPal実売上

## 参考資料
- [GMO-PG PayPal API 都度決済仕様書](https://gmopg_docs:PF%cwa$GmCC@docs.mul-pay.jp/paypal/payg-api)

wamazing-sei-sato committed:
PayPal決済機能を追加

・取引登録、決済実行、キャンセルメソッドを実装し、定数にPayPalの支払同意IDを追加

wamazing-sei-sato commented:
@Roy-Mao @wujietao-wamazing 
お疲れさまです。
PayPal決済対応のため、クライアントライブラリを更新しました。
お手すきの際にご確認いただけると助かります 🙇

wujietao-wamazing reviewed (approved):
@wamazing-sei-sato 
自分にはLGTMです。
お疲れ様でした。

wamazing-sei-sato committed:
feat: PayPal決済の仮売上取消・実売上APIに対応

支払方法を保存するアーキテクチャから都度支払の方式に対応できるよう、
PayPal決済に以下のAPIメソッドを追加：

- cancel_auth_paypal: 仮売上取消 (/payment/CancelAuthPaypal.idPass)
- paypal_sales: 実売上 (/payment/PaypalSales.idPass)

これにより、PayPal決済でも都度決済フローが利用可能になる。

See: wamazing/wamazing#3486

wamazing-sei-sato commented:
@wujietao-wamazing @Roy-Mao 
お疲れさまです。
都度支払方式への変更に伴い、GMOクライアントライブラリを修正しました。

変更点: [https://github.com/wamazing/gmo-payment/pull/7/commits/205643880b4cc615864bea4a011f80bf9ee0abb3](https://github.com/wamazing/gmo-payment/pull/7/commits/205643880b4cc615864bea4a011f80bf9ee0abb3)

後ほどMTGでも共有いたします 🙇

wamazing-sei-sato commented:
ご確認ありがとうございました 🙇
マージさせていただきます。
